### PR TITLE
feat: GO Feature Flag provider

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR
GO Feature Flag golang provider.
Tests are not passing because of this issue https://github.com/open-feature/go-sdk/issues/117.

### CODE REVIEW

File: providers/go-feature-flag/pkg/provider.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+		if err != nil {
+			return nil, err
+		}
+		return &Provider{
+			goFeatureFlagInstance: goff,
+		}, nil
+	}
+
+	if options.Endpoint == "" {
+		return nil, fmt.Errorf("invalid provider options, invalid endpoint value: %s", options.Endpoint)
```
Comment (skyerus) [inline providers/go-feature-flag/pkg/provider.go:58]:
```suggestion
		return nil, fmt.Errorf("invalid provider options, empty endpoint value")
```

File: providers/go-feature-flag/pkg/provider.go [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+	// if we have a GO Feature Flag instance instantiate we evaluate the flag locally,
+	// using the GO module directly. We will not send any remote calls to the relay proxy.
+	if provider.goFeatureFlagInstance != nil {
+		return evaluateLocally(provider, goffRequestBody, flagName, defaultValue)
+	}
+	return evaluateWithRelayProxy(provider, ctx, goffRequestBody, flagName, defaultValue)
+}
+
+// evaluateLocally is using the GO Feature Flag module to evaluate your flag.
+// it means that you don't need any rela proxy to make it works.
```
Comment (skyerus) [inline providers/go-feature-flag/pkg/provider.go:142]:
```suggestion
// it means that you don't need any relay proxy to make it work.
```

File: providers/go-feature-flag/pkg/provider.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+	if err != nil {
+		return model.GenericResolutionDetail[T]{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewGeneralResolutionError("impossible to contact GO Feature Flag relay proxy instance"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	responseStr, err := ioutil.ReadAll(response.Body)
```
Comment (skyerus) [inline providers/go-feature-flag/pkg/provider.go:290]: `ioutil.ReadAll` is deprecated, use `io.ReadAll` https://pkg.go.dev/io/ioutil#ReadAll

File: providers/go-feature-flag/pkg/provider.go [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+			return model.GenericResolutionDetail[T]{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewParseErrorResolutionError(
+						fmt.Sprintf("impossible to parse response for flag %s: %s", flagName, responseStr)),
+					Reason: of.ErrorReason,
+				},
+			}
+		}
+		fmt.Println(err)
```
Comment (skyerus) [inline providers/go-feature-flag/pkg/provider.go:314]: should this be here?

File: providers/go-feature-flag/pkg/provider_options.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package gofeatureflag
+
+import ffclient "github.com/thomaspoignant/go-feature-flag"
+
+// ProviderOptions is the struct containing the provider options you can
+// use while initializing GO Feature Flag.
+// To have a valid configuration you need to have at Endpoint or GOFeatureFlagConfig to be set.
```
Comment (skyerus) [inline providers/go-feature-flag/pkg/provider_options.go:7]:
```suggestion
// To have a valid configuration you need to have an Endpoint or GOFeatureFlagConfig set.
```

File: providers/go-feature-flag/pkg/provider.go (hunk 4 above)
Comment (thomaspoignant) [reply providers/go-feature-flag/pkg/provider.go:314]: oh good catch I missed this one.

**Review by skyerus — APPROVED** [review]
LGTM :)

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**